### PR TITLE
Import rust-guidelines

### LINKED ISSUE #19315: Publish guidelines somewhere official
The coding guidelines are currently hosted at @aturon's github page: http://aturon.github.io/. These should be hosted somewhere official. Requires some network setup and possibly automation since there's a build step.

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #19315

r? @aturon

### CODE REVIEW

**Comment by aturon** [comment]
Awesome!

One question: I've never been overly fond of "Rust Guidelines" for the title of this, although now that the "Guides" are not a thing it's a little better. Any bikeshedding ideas?

**Comment by steveklabnik** [comment]
PEP 8 is called "Style guide for python code". So "Style Guide"?

**Comment by aturon** [comment]
OK, sounds fine to me. Would you be willing to do the rename now, before we bring it in?

**Comment by steveklabnik** [comment]
@aturon done, I think. :smile:

File: src/doc/style/.gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Generated when running `gitbook build`
+_book/
+
+*~
+*.bak
+*.swp
+.DS_Store
```
Comment (alexcrichton) [inline src/doc/style/.gitignore]: This could probably be left out

File: src/doc/style/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+% Rust Guidelines [working title]
```
Comment (alexcrichton) [inline src/doc/style/README.md]: Style Guide?

File: src/doc/style/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+[some infrastructure work](https://github.com/aturon/rust-guidelines/issues/17), the snapshot will move somewhere more
+official.
+
+### Building the document
+
+Like http://rustbyexample.com/, this guidelines document is written in
+the [`gitbook`](https://github.com/GitbookIO/gitbook) style. It can be
+compiled with a prototype tool,
+[`rustbook`](https://github.com/aturon/rust-book) that provides a
+minimal subset of `gitbook`'s functionality on top of `rustdoc`.
```
Comment (alexcrichton) [inline src/doc/style/README.md]: Hm, this probably needs to be edited and it may need a general pass as well in more places, but that may be able to wait.

Comment (alexcrichton) [inline src/doc/style/README.md]: Eh actually maybe there isn't that much else!

File: src/doc/style/README.md (hunk 2 above)
Comment (steveklabnik) [inline src/doc/style/README.md]: dammit, I knew i missed something

**Comment by steveklabnik** [comment]
@alexcrichton nice, thanks! Fixed.

File: src/doc/style/.gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+*~
+*.bak
+*.swp
```
Comment (alexcrichton) [inline src/doc/style/.gitignore]: er I meant could this whole file be remove?

**Comment by alexcrichton** [comment]
r=me with one small nit

I noticed that this is not actually linked from any location, but I think that's a good thing for now. We should take some more time to flesh out these documents before we're ready to link to them from the main page.

**Comment by steveklabnik** [comment]
re links: exactly

**Comment by steveklabnik** [comment]
re links: exactly what i was thinking.

**Pull request merged**